test/cpp/end2end/xds_end2end_test.cc

```diff
@@ -2195,18 +2230,31 @@ TEST_P(LdsTest, ChooseMatchedDomain) {
       ->mutable_routes(0)
       ->mutable_route()
       ->mutable_cluster_header();
-  balancers_[0]->ads_service()->SetLdsResource(
-      AdsServiceImpl::BuildListener(route_config), kDefaultResourceName);
+  if (GetParam().enable_rds_testing()) {
```

Comment:
I suggest creating a helper method in `XdsEnd2endTest` for this:

```
void SetRouteConfiguration(int idx, const RouteConfiguration& route_config) {
  AdsServiceImpl* ads_service = balancers_[idx]->ads_service();
  if (GetParam().enable_rds_testing()) {
    ads_service->SetRdsResource(route_config, kDefaultResourceName);
  } else {
    ads_service->SetLdsResource(AdsServiceImpl::BuildListener(route_config), kDefaultResourceName);
  }
}
```

That way, the code in each test can just say this:

```
SetRouteConfiguration(0, route_config);
```